src/services/scope-manager.js

```diff
@@ -15,14 +15,28 @@ class ScopeManager {
     this.scopesCache = {};
   }
 
+  async appendScopeForUser(previousFilter, user, collectionName) {
+    const newFilter = await this.getScopeForUser(user, collectionName, true);
+    const filters = [previousFilter, newFilter].filter(Boolean);
```

Comment:
[Naming] can't we find better names for `previous` and `new` filters? something like `existingFilter` and `scopeFilter` maybe?